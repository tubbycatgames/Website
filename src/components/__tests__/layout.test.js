import React from "react"
import renderer from "react-test-renderer"

import mockSiteQuery from "../../test-utils/mock-site-query"

import Layout from "../layout"

describe("Layout", () => {
  beforeEach(() => {
    mockSiteQuery()
  })

  it("renders all site sections and children", () => {
    // Avoids a deprecation warning from helmet
    console.warn = jest.fn()

    const tree = renderer.create(
      <Layout>
        <div>
          <t>Some Layout Content</t>
        </div>
      </Layout>
    )
    expect(tree.toJSON()).toMatchSnapshot();
  })
})
