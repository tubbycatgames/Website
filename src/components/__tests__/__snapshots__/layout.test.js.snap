// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders all site sections and children 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <div
      className="icons"
    >
      <a
        href="https://tubbycatgames.com"
        onClick={[Function]}
      >
        <img
          alt="Tubby Cat Games Icon"
          className="logo"
          src="test-file-stub"
        />
      </a>
      <a
        href="https://tubbycatgames.itch.io/"
        onClick={[Function]}
      >
        <img
          alt="Itch Page Icon"
          className="icon"
          src="test-file-stub"
        />
      </a>
      <a
        href="https://gamejolt.com/@tubbycatgames"
        onClick={[Function]}
      >
        <img
          alt="Game Jolt Page Icon"
          className="icon"
          src="test-file-stub"
        />
      </a>
      <a
        href="https://github.com/tubbycatgames"
        onClick={[Function]}
      >
        <img
          alt="GitHub Page Icon"
          className="icon"
          src="test-file-stub"
        />
      </a>
      <a
        href="https://discord.gg/vyh5XY"
        onClick={[Function]}
      >
        <img
          alt="Discord Social Media Icon"
          className="icon"
          src="test-file-stub"
        />
      </a>
      <a
        href="https://twitter.com/tubbycatgames"
        onClick={[Function]}
      >
        <img
          alt="Twitter Social Media Icon"
          className="icon"
          src="test-file-stub"
        />
      </a>
      <a
        href="https://instagram.com/tubbycatgames"
        onClick={[Function]}
      >
        <img
          alt="Instagram Social Media Icon"
          className="icon"
          src="test-file-stub"
        />
      </a>
      <a
        href="https://fb.me/tubbycatgames"
        onClick={[Function]}
      >
        <img
          alt="Facebook Social Media Icon"
          className="icon"
          src="test-file-stub"
        />
      </a>
      <a
        href="https://www.youtube.com/channel/UC_6l3SdhVjdtTd2DnHK2JbQ"
        onClick={[Function]}
      >
        <img
          alt="YouTube Channel Icon"
          className="icon"
          src="test-file-stub"
        />
      </a>
    </div>
    <div
      className="links"
    >
      <a
        href="/"
      >
        <p>
          About
        </p>
      </a>
      <a
        href="/games"
      >
        <p>
          Games
        </p>
      </a>
    </div>
  </div>
  <div
    className="content"
  >
    <div>
      <t>
        Some Layout Content
      </t>
    </div>
  </div>
  <div
    className="footer"
  >
    <div>
      <h2
        className="header"
      >
        Join our mailing list for game updates!
      </h2>
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          aria-label="Mailing List Signup Email"
          className="input"
          name="email"
          onChange={[Function]}
          placeholder="Enter your email here"
          required={true}
          type="email"
          value=""
        />
        <button
          className="button"
          type="submit"
        >
          Sign up!
        </button>
      </form>
    </div>
  </div>
</div>
`;
